omnisharp.el

```diff
@@ -277,6 +277,59 @@ asynchronously. On completions, CALLBACK is run with the quickfixes as its only
      (apply callback (list (omnisharp--vector-to-list
                             (cdr (assoc 'QuickFixes quickfix-response))))))))
 
+(defun omnisharp-find-implementations-popup ()
+  "Show a popup containing all implementations of the interface under
+point, or classes derived from the class under point. Allow the user
+to select one (or more) to jump to."
+  (interactive)
+  (message "Finding implementations...")
+  (omnisharp-find-implementations-worker
+   (omnisharp--get-common-params)
+   (lambda (quickfixes)
+     (cond ((equal 0 (length quickfixes))
+	    (message "No implementations found."))
+
+	   ;; Go directly to the implementation if there only is one
+	   ((equal 1 (length quickfixes))
+	    (omnisharp-go-to-file-line-and-column (car quickfixes)))
+
+	   (t
+	    (omnisharp-navigate-to-implementations-popup quickfixes))))))
+
+(defun omnisharp-get-implementation-title (item)
+  "Get the human-readable class-name declaration from an alist with
+information about implementations found in omnisharp-find-implementations-popup."
+  (let* ((text (cdr (assoc 'Text item))))
+    (if (or (string-match-p " class " text)
+	      (string-match-p " interface " text))
+	text
+      (concat
+       (file-name-nondirectory (cdr (assoc 'FileName item)))
+       ":"
+       (number-to-string (cdr (assoc 'Line item))))
+      )))
+
+(defun omnisharp-get-implementation-titles (items)
+  "Get a list of the human-readable class-name declaration from a list
+implementations found in omnisharp-find-implementations-popup."
+  (mapcar 'omnisharp-get-implementation-title items))
+
+(defun omnisharp-get-implementation-by-name (items title)
+  "Return the implementation-object which matches the provided title."
```

Comment:
I think this would be better done with a higher order function:

```lisp
(--first (string= title (omnisharp-get-implementation-title it))
         items)
```
See: https://github.com/magnars/dash.el#-first-pred-list